src/transformStateWithClones.js

```diff
@@ -1,13 +1,56 @@
 'use strict';
 
 /**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
+ * Deep clone function to create a copy of an object.
+ * @param {Object} obj - The object to be cloned.
+ * @returns {Object} - The deep clone of the object.
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [simpleClone(state)];
+
+  for (const action of actions) {
+    let currentStateCopy = simpleClone(resultArray[resultArray.length - 1]);
+
+    switch (action.type) {
+      case 'addProperties':
```

Comment:
Describe all variants of cases in constant.
const ACTION_CASES = {
add: 'addProperties',
}